backend/src/main/java/com/handong/rebon/shop/domain/category/Category.java

```diff
@@ -2,22 +2,50 @@
 
 import javax.persistence.*;
 
-import com.handong.rebon.shop.domain.item.Shops;
+import com.handong.rebon.exception.category.CategoryNameException;
+
+import lombok.*;
+
+import java.util.List;
 
 @Entity
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+@AllArgsConstructor
+@Getter
+@Setter
 public class Category {
 
-    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
 
     private String name;
 
-    @Embedded
-    private Shops shops;
+    @OneToMany(mappedBy = "category", fetch = FetchType.LAZY)
+    private List<ShopCategory> shopCategories;
 
     @ManyToOne
     private Category parent;
 
     @Embedded
     private Categories children;
+
+    @Builder
+    public Category(String name, List<ShopCategory> shopCategories, Category parent, Categories children) {
+        validatesBlankName(name);
+        this.name = name;
+        this.parent = parent;
+        this.children = children;
+    }
+
+    private void validatesBlankName(String name) {
+        if (name.isBlank()) {
+            throw new CategoryNameException();
+        }
+    }
+
+    public void addChildCategory(Category category){
+        this.getChildren().addChild(category);
```

Comment:
굳이 getChildren하지 않고 children에 접근할 수 있지 않나요??
```suggestion
        this.children.addChild(category);
```

Reply:
음 사실 맞습니다. 제가 getter를 사용하는것이 직접 필드에 접근하는 것보다 안전하다는 말을 들어서 근데 어차피 클래스내부라 이때의 경우가 아니였던거 같습니다. 수정하겠습니다!